utensor_cgen/backend/transformer.py

```diff
@@ -0,0 +1,74 @@
+import pickle
+
+from utensor_cgen.logger import logger
+from utensor_cgen.transformer.pipeline import TransformerPipeline
+from utensor_cgen.utils import class_property
+
+from .base import BackendPart
+
+__all__ = ['PipelineTransformer']
+
+class PipelineTransformer(BackendPart):
+  TARGET = 'generic'
+  PART = 'pipeline_transformer'
+
+  def __init__(self, config):
+    self.transformer = TransformerPipeline(
+      methods=self.config['transform_methods']
+    )
+    self.trans_methods = self.config['transform_methods']
+    self.save_graph = self.config['save_graph']
+
+  def transform(self, ugraph):
+    logger.info("Transforming graph: %s", ugraph.name)
+    logger.info("Transform pipeline: %s", ' -> '.join(self.trans_methods))
+    self._check_non_quantized(ugraph)
+    new_ugraph = self.transformer.transform(ugraph)
+    new_ugraph.name = ugraph.name
+    logger.info('Graph transormation done')
+    if self.save_graph:
+      logger.info('Saving transformed graph')
+      pkl_fname = "{}_transformed.pkl".format(ugraph.name)
+      with open(pkl_fname, 'wb') as fid:
+        pickle.dump(new_ugraph, fid)
+      logger.info('{} saved'.format(pkl_fname))
+    return new_ugraph
+  
+  @classmethod
+  def _check_non_quantized(cls, ugraph):
+    is_quantized = False
+    quant_ops = set([
+      "Dequantize", "QuantizedMaxPool",
```

Comment:
These will be the uTensor equivalent names correct?

Reply:
Well, yes.
I'm thinking we will need to formally define a set of op names for quantization (generic names).
We don't have such generic ops yet but it's not hard to fix it once we do.

Reply:
We can always go off of the ops in the runtime, for naming requirements, in/outs, params, and additional attributes. Then for uTensor targets the conversion is a nop, for any other targets we just do some finagling

Reply:
Can you elaborate more about go off the ops in the runtime for naming requirements, in/outs, params and attributes? I don't understand what you mean.
An example will help.